packages/calcite-components/src/components/list/list.tsx

```diff
@@ -844,39 +850,50 @@ export class List
     }));
   };
 
-  private updateListItems = debounce((emitFilterChange = false): void => {
-    const { selectionAppearance, selectionMode, dragEnabled, el } = this;
-
-    const items = Array.from(this.el.querySelectorAll(listItemSelector));
-
-    items.forEach((item) => {
-      item.selectionAppearance = selectionAppearance;
-      item.selectionMode = selectionMode;
-      if (item.closest("calcite-list") === el) {
-        item.dragHandle = dragEnabled;
+  private updateListItems = debounce(
+    ({
+      emitFilterChange = false,
+      performFilter = false,
+    }: {
+      emitFilterChange: boolean;
+      performFilter: boolean;
+    }): void => {
+      const { selectionAppearance, selectionMode, dragEnabled, el, filterEl, filterEnabled } = this;
+
+      const items = Array.from(this.el.querySelectorAll(listItemSelector));
+
+      items.forEach((item) => {
+        item.selectionAppearance = selectionAppearance;
+        item.selectionMode = selectionMode;
+        if (item.closest("calcite-list") === el) {
+          item.dragHandle = dragEnabled;
+        }
+      });
+
+      if (this.parentListEl) {
+        this.setUpSorting();
+        return;
       }
-    });
 
-    if (this.parentListEl) {
-      this.setUpSorting();
-      return;
-    }
+      this.listItems = items;
+      if (filterEnabled && performFilter) {
+        this.dataForFilter = this.getItemData();
 
-    this.listItems = items;
-    if (this.filterEnabled) {
-      this.dataForFilter = this.getItemData();
-      if (this.filterEl) {
-        this.filterEl.items = this.dataForFilter;
+        if (filterEl) {
+          filterEl.items = this.dataForFilter;
+          this.filterAndUpdateData();
```

Comment:
Does this need to be awaited on to avoid synchronization issues?

Reply:
@jcfranco It doesn't. The filter component will fire an event once the filter actually occurs.

This is part of the oddness of having a filter component that does the filtering. The syncing is odd.